users/adapter.py

```diff
@@ -10,6 +10,10 @@ def get_email_confirmation_url_stub(self, request, emailconfirmation):
         return settings.SSL_FRONTEND_HOST + url
 
     def clean_email(self, email):
+        if len(settings.ALLOWED_DOMAINS_FOR_USER_REGISTRATION):
```

Comment:
I'd suggest to query for this more defensively, eg like this
```

if hasattr(settings, "ALLOWED_DOMAINS_FOR_USER_REGISTRATION") ...
```

this ensures that `settings` won't break with an `AttributeError` in case it's missing in settings.py

Also no need to check with len() explicitly, the same can be through:

```
if hasattr(settings, "ALLOWED_DOMAINS_FOR_USER_REGISTRATION") and settings.ALLOWED_DOMAINS_FOR_USER_REGISTRATION: 
```